### PR TITLE
Create agent cluster secret

### PR DESCRIPTION
**Pull request by pfmackin**
# Description

As part of the work to have the admin cluster also be a managed cluster, we need to have the agent and MC controllers use different secrets.  The MC agent on the managed cluster will watch for the agent secret (which will only be on a  managed cluster) then start fetching MC resources from the admin cluster.  The MC controllers get the managed cluster identify from the registration secret. We don't want the admin cluster to use the agent since the MC resources are already local.   This change allows the agent to be idle on the admin cluster, while the MC controllers on the admin cluster work exactly like they do on a managed cluster.

Also, minor renaming to make the sync files consistent.  Remove the registration secret field from VMC since it is not needed.

Fixes VZ-2213

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/clusters/sync_registration_secret.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	_, err := r.createOrUpdateRegistrationSecret(vmc, secretName, managedNamespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Create or update the kubeconfig secret
```
Comment (mgianatagh) [inline platform-operator/controllers/clusters/sync_registration_secret.go:27]: the comment mentions updating the kubeconfig secret, it looks like only a cluster name is being updated

File: platform-operator/controllers/clusters/vmc_controller_test.go [hunk 2]
```diff
@@ -167,14 +169,61 @@ func fakeGetConfig() (*rest.Config, error) {
...
+	mock.EXPECT().
+		Get(gomock.Any(), types.NamespacedName{Namespace: "", Name: generateManagedResourceName(name)}, gomock.Not(gomock.Nil())).
+		Return(errors.NewNotFound(schema.GroupResource{Group: "", Resource: "ServiceAccount"}, generateManagedResourceName(name)))
+
+	// Expect a call to create the ClusterRoleBinding - return success
+	mock.EXPECT().
+		Create(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx context.Context, binding *rbacv1.ClusterRoleBinding, opts ...client.CreateOption) error {
+			asserts.Equalf(generateManagedResourceName(name), binding.Name, "ClusterRoleBinding name did not match")
+			asserts.Equalf("verrazzano-managed-cluster", binding.RoleRef.Name, "ClusterRoleBinding roleref did not match")
```
Comment (desagar) [inline platform-operator/controllers/clusters/vmc_controller_test.go:186]: is there a constant for the ClusterRole name of "verrazzano-managed-cluster"?

File: application-operator/mcagent/mcagent.go [hunk 3]
```diff
@@ -22,13 +22,13 @@ import (
...
+	// Wait for the existence of the verrazzano-cluster-agent secret.  It contains the credentials
 	// for connecting to a managed cluster.
 	log.Info("Starting multi-cluster agent")
 	secret := corev1.Secret{}
 
 	for {
-		err := client.Get(context.TODO(), types.NamespacedName{Name: constants.MCRegistrationSecret, Namespace: constants.VerrazzanoSystemNamespace}, &secret)
+		err := client.Get(context.TODO(), types.NamespacedName{Name: constants.MCAgentSecret, Namespace: constants.VerrazzanoSystemNamespace}, &secret)
 		if err == nil {
 			err := validateClusterSecret(&secret)
```
Comment (desagar) [inline application-operator/mcagent/mcagent.go:33]: rename to validateAgentSecret?

File: platform-operator/controllers/clusters/vmc_controller_test.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/vmc_controller_test.go:186]: let me add one

File: application-operator/mcagent/mcagent.go (hunk 3 above)
Comment (pfmackin) [reply application-operator/mcagent/mcagent.go]: fixed

File: platform-operator/controllers/clusters/vmc_controller_test.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/vmc_controller_test.go]: fixed

File: platform-operator/controllers/clusters/sync_registration_secret.go (hunk 1 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/sync_registration_secret.go]: fixed

**Review by mgianatagh — DISMISSED** [review]

**Review by desagar — DISMISSED** [review]

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**